packages/quantic/cypress/integration/facets/category-facet/catgory-facet.cypress.ts

```diff
@@ -0,0 +1,200 @@
+import {configure} from '../../../page-objects/configurator';
+
+import {CategoryFacetExpectations as Expect} from './category-facet-expectations';
+import {
+  canadaHierarchy,
+  canadaHierarchyIndex,
+  CategoryFacetActions as Actions,
+} from './category-facet-actions';
+import {InterceptAliases, interceptSearch} from '../../../page-objects/search';
+
+interface CategoryFacetOptions {
+  field: string;
+  label: string;
+  basePath: string;
+  noFilterByBasePath: boolean;
+  noFilterFacetCount: boolean;
+  delimitingCharacter: string;
+  numberOfValues: number;
+  sortCriteria: string;
+  withSearch: boolean;
+  isCollapsed: boolean;
+}
+
+describe('quantic-category-facet', () => {
+  const pageUrl = 's/quantic-category-facet';
+
+  const defaultField = 'geographicalhierarchy';
+  const defaultLabel = 'Country';
+  const defaultNumberOfValues = 8;
+
+  function visitCategoryFacetPage(
+    options: Partial<CategoryFacetOptions> = {},
+    waitForSearch = true
+  ) {
+    interceptSearch();
+    cy.visit(pageUrl);
+    configure(options);
+    if (waitForSearch) {
+      cy.wait(InterceptAliases.Search);
+    }
+  }
+  function setupWithDefaultSettings() {
+    visitCategoryFacetPage(
+      {
+        field: defaultField,
+        label: defaultLabel,
+        numberOfValues: defaultNumberOfValues,
+      },
+      false
+    );
+  }
+
+  function setupGoDeeperOneLevel() {
+    setupWithDefaultSettings();
+    Actions.selectChildValueAt(canadaHierarchyIndex[0]);
+  }
+
+  function setupGoDeeperTwoLevels() {
+    setupWithDefaultSettings();
+    Actions.selectChildValueAt(canadaHierarchyIndex[0]);
+    interceptSearch();
+    Actions.selectChildValueAt(canadaHierarchyIndex[1]);
+    interceptSearch();
+  }
+
+  function setupShowMore() {
+    setupWithDefaultSettings();
+
+    setupGoDeeperOneLevel();
+    Actions.clickShowMoreButton();
+  }
+
+  function setupShowLess() {
+    setupShowMore();
+    interceptSearch();
+    Actions.clickShowLessButton();
+  }
+
+  function setupGoDeeperFourLevels() {
+    setupWithDefaultSettings();
+    Actions.selectChildValueAt(canadaHierarchyIndex[0]);
+    interceptSearch();
+    Actions.selectChildValueAt(canadaHierarchyIndex[1]);
+    interceptSearch();
+    Actions.selectChildValueAt(canadaHierarchyIndex[2]);
+    interceptSearch();
+    Actions.selectChildValueAt(canadaHierarchyIndex[3]);
+    interceptSearch();
+  }
+
+  describe('with default category facet', () => {
+    it('should work as expected', () => {
+      setupWithDefaultSettings();
+
+      Expect.isAccesssible(true);
+      Expect.displayLabel(true);
+      Expect.displayFacetCount(true);
+      Expect.displaySearchInput(false);
+      Expect.numberOfValues(7);
```

Comment:
I like how readable it is, however it would be a tad easier to maintain by replacing `7` by `defaultNumberOfValues -  1`.

Just to be sure, we have to remove `1` from the count because of the `All` option, right?